### ISSUE TITLE
[create] dynamic readmes for apps

### ISSUE THREAD

**Issue by thepassle** [open-wc/open-wc#959]
Currently, the dynamic readmes only work when scaffolding a single component. We should add it to apps too

See:
https://github.com/open-wc/open-wc/pull/919/files#diff-a8d1b3e29b78bbefc0dee463468fe67fR1

**Comment by bennypowers** [comment]
Agreed. I think the dynamic README stuff should be higher up the inheritance chain in Generator.js
I'll work on this a little. Would be nice to get #941 in first.

**Comment by bennypowers** [comment]
We discussed this a little on slack.

The plan so far is to 
1. merge #976 to set up at least one basic integration test (the more, the merrier)
1. refactor the generator code to try and separate out the pure (logic) and impure (side-effecting, i/o) sides, starting by moving the CLI code out of the Generator class inheritance graph
1. generalize the template-partial code to work horizontally (across features) and vertically (up and down the inheritance graph, e.g. when recursing from app into app-component)
